lib/core/waitForCoreChainLockSyncFactory.js

```diff
@@ -0,0 +1,70 @@
+const { ChainLock } = require('@dashevo/dashcore-lib');
+const ZMQClient = require('@dashevo/dashd-zmq');
+const waitForEvent = require('../util/waitForEvent');
+const ensureBlock = require('../util/ensureBlock');
+
+/**
+ * Check that core is synced (factory)
+ *
+ * @param {ZMQClient} coreZMQClient
+ * @param {RpcClient} coreRpcClient
+ * @param {LatestCoreChainLock} latestCoreChainLock
+ * @param {Logger} logger
+ *
+ * @returns {waitForCoreSync}
+ */
+function waitForCoreChainLockSyncFactory(
+  coreZMQClient,
+  coreRpcClient,
+  latestCoreChainLock,
+  logger,
+) {
+  /**
+   * Check that chainLock is synced
+   *
+   * @typedef waitForCoreChainLockSync
+   *
+   * @returns {Promise<LatestCoreChainLock>}
+   */
+  async function waitForCoreChainLockSync() {
+    await coreZMQClient.connect();
+    // listens to ChainLock zmq zmqpubrawchainlocksig event and updates the latestCoreChainLock.
+    // If we lost connection with Core (heartbeats with retry policy) we need to exit with fatal
+    // In case of regtest fallback we need to listen for new blocks using hashblock zmq event
+    coreZMQClient.subscribe(ZMQClient.TOPICS.rawchainlock);
+    logger.info('Subscribe to rawchainlock ZMQ room');
+
+    coreZMQClient.on(ZMQClient.TOPICS.rawchainlock, async (rawChainLockMessage) => {
+      const socketChainLock = new ChainLock(rawChainLockMessage);
+      logger.info(`Received new chainlock ${socketChainLock.toString()}`);
+      await ensureBlock(coreZMQClient, coreRpcClient, socketChainLock.blockHash);
+      latestCoreChainLock.update(socketChainLock);
+    });
+
+    // Because a Chainlock may happen before its block, we also subscribe to rawblock
+    coreZMQClient.subscribe(ZMQClient.TOPICS.hashblock);
+    logger.info('Subscribe to hashblock ZMQ room');
```

Comment:
Do you think it could help?

Reply:
As `info` may be not so much, as a `trace` yeah, it's a dependency, if you have an issue at this point, trace might highlight the where (i'm pretty pro-verbose as trace tbh).

Reply:
Sure! let's change the logging level for this output!

Reply:
Agreed to use `trace`